### ISSUE TITLE
[Bug]: Leave GTA online not work

### ISSUE THREAD

**Issue by BeztNA** [Yimura/YimMenu#1201]
### Describe the bug

Leave GTA online not work

### Steps To Reproduce

1.Long loading
2.Leave GTA online

### Expected Behavior

work

### Screenshots and/or Logs

```
<paste your logs here>
```

### Store

Steam

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

**Comment by Yimura** [comment]
The leave GTA online button you're talking about doesn't work while you're in a loading screen.
All this button does is set some values in memory to tell the game to switch session (or leave it and go to single player).
The game ignores these changes in memory until you're completely loaded into the session.

https://github.com/YimMenu/YimMenu/blob/51e3b6e132908143902ee9d01703f1f12388abdb/src/util/session.hpp#L36-L48

If you really must leave online no matter what you can use the Network Bail button which can be found under Settings -> Debug -> Misc -> Network Bail

https://github.com/YimMenu/YimMenu/blob/51e3b6e132908143902ee9d01703f1f12388abdb/src/views/debug/view_debug_misc.cpp#L34-L36

**Comment by BeztNA** [comment]
> The leave GTA online button you're talking about doesn't work while you're in a loading screen. All this button does is set some values in memory to tell the game to switch session (or leave it and go to single player). The game ignores these changes in memory until you're completely loaded into the session.
> 
> https://github.com/YimMenu/YimMenu/blob/51e3b6e132908143902ee9d01703f1f12388abdb/src/util/session.hpp#L36-L48
> 
> If you really must leave online no matter what you can use the Network Bail button which can be found under Settings -> Debug -> Misc -> Network Bail
> 
> https://github.com/YimMenu/YimMenu/blob/51e3b6e132908143902ee9d01703f1f12388abdb/src/views/debug/view_debug_misc.cpp#L34-L36

but in kiddion, i can leave online in loading screen

**Comment by Yimura** [comment]
Yeah I have no idea how Kiddion's code does it to leave online.
As I said, you can use the button I mentioned to force leave GTA for now, I'll take a look at improving it.

**Comment by BeztNA** [comment]
Thank you

**Comment by Yimura** [comment]
Kindly keep the issue open for as long as there is no fix implemented.

**Comment by CreepyQuietWeirdo** [comment]
FWIW, the "Leave GTA Online" pause menu option is implemented in `pausemenu_multiplayer::func_46`, in switch case 15:
```
		case 15:
			func_85(1);
			func_81(1);
			if (NETWORK::NETWORK_IS_ACTIVITY_SESSION()) {
				func_259();
			}
			func_122();
			func_119();
			func_86(1, 1);
			func_269();
			func_260(iVar20);
			func_83(3);
			NETWORK::RESET_STORE_NETWORK_GAME_TRACKING();
			func_281();
			func_274(-1);
			func_79(14);
			break;
```
Leaving is not really a join type, a couple more globals are required if you want it to work while loading is in progress. If you mimic the code in pausemenu_multiplayer, the leave option could become even more reliable than the one modest menu is using.

**Comment by Yimura** [comment]
Possibly fixed by #1725
